### DISCUSSION TITLE
Entity Definitions Endpoint

### DISCUSSION THREAD

**Discussion by Gareth064** [microsoft/PowerPlatform-DataverseServiceClient#484] [Q&A]
Hi

How can I call `https://org.crm.dynamics.com/api/data/v9.2/EntityDefinitions` using this SDK?

I am looking to build an AI tool which will need to know the entity schemas of some tables.

Thanks.

**Comment by JimDaly** [comment]
See [Query schema definitions](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/query-schema-definitions?view=dataverse-latest&tabs=sdk) which explains how to use the powerful `RetrieveMetadataChanges` message ([RetrieveMetadataChangesRequest Class](https://learn.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.messages.retrievemetadatachangesrequest?view=dataverse-sdk-latest)). It includes a table in [Evaluate other options to retrieve schema definitions](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/query-schema-definitions?view=dataverse-latest&tabs=sdk#evaluate-other-options-to-retrieve-schema-definitions) that compares different options available.

Not only does `RetrieveMetadataChanges` provide powerful ways to filter tables, columns, and relationships, you can use it to maintain a cache more efficiently.
